### DISCUSSION TITLE
Reuse of Jaeger UI Components in other Applications

### DISCUSSION THREAD

**Discussion by zhuje** [jaegertracing/jaeger#4521] [Q&A] [answered]
Hello All, 

Is there a way to import Jager UI Components so I can reuse them in my application(e.g., something similar to npm packages where I can import individual components)? Particularly, I wanted to reuse the gantt chart and scatterplot. 

I saw that there is an 'Embedded Mode' offered (https://www.jaegertracing.io/docs/1.46/frontend-ui/#embedded-mode). However, my goal is to skin the components, so they look similar to the theme of my application. 

There was some previous discussions on this topic from 2021 (https://github.com/jaegertracing/jaeger-ui/issues/507) and I was wondering if there were any more thoughts on this.

**Comment by yurishkuro** [comment, marked as answer]
I am not a front expert, but I assume at this time the answer is No even simply because we don't publish the components anywhere. I expect there needs to be a different project structure. We already have the graph viz separated into a `plexus` component that is technically usable independently of Jaeger UI.

I have no insights as to the skinning capabilities. There were some PRs done by Grafana team at some point, but they gave up and forked the code instead.

If you have suggestions how this can be done, please share.
